### DISCUSSION TITLE
A question about token refreshing with oauth.v2.access

### DISCUSSION THREAD

**Discussion by WZRPW** [slack-go/slack#1008] [Q&A] [answered]
Follow the doc: https://api.slack.com/authentication/rotation#refresh
Question:
1. does the refresh token have an expiring date/time?The API response only has expire_in for the access token. 
2. When using the refresh token to acquire a new access token, the response will contain a new refresh token. Will the old refresh token be in-validated (or expire) after the token-refresh API call?
Thanks,

**Comment by seratch** [comment, marked as answer]
>1. does the refresh token have an expiring date/time? The API response only has expire_in for the access token.

Until you use the refresh token for rotating the access token, the refresh token can be valid.

> 2. When using the refresh token to acquire a new access token, the response will contain a new refresh token. Will the old refresh token be in-validated (or expire) after the token-refresh API call?

The current one and its previous generation can be valid. When you rotate tokens again with the refresh token and access token, the old one will expire.
